builder/amazon/common/step_security_group.go

```diff
@@ -45,7 +45,9 @@ func (s *StepSecurityGroup) Run(state multistep.StateBag) multistep.StepAction {
 
 	port := s.CommConfig.Port()
 	if port == 0 {
-		panic("port must be set to a non-zero value.")
+		if s.CommConfig.Type != "none" {
+			panic("port must be set to a non-zero value.")
```

Comment:
hmm fair points all around. I think running the create sg step with a nil communicator is still probably not the right thing to do, because it's not going to be testing anything, really, but it still won't hurt to just have it in there with a dummy value. These lesser documented edge-cases can have a few warts.

Have you successfully built the nil communicator with this patch?